pkgs/development/compilers/gcc/7/default.nix

```diff
@@ -88,7 +88,10 @@ let version = "7.3.0";
         # maybe only needed on musl, PATH_MAX
         # https://github.com/richfelker/musl-cross-make/blob/0867cdf300618d1e3e87a0a939fa4427207ad9d7/litecross/Makefile#L62
         "--disable-libmpx"
-      ] else [
+      ] ++ stdenv.lib.optionals ( targetPlatform.config == "powerpc64le-unknown-linux-gnu" )  [
+	"--with-long-double-128"
+	"--enable-softfloat"
```

Comment:
See what I did with the ABIs in the nixpkgs lib to lessen the cominatorial explosion in GCC. Can you use the existing functionality for soft floats, and make a new one for longer floats?